azext_iot/assets/user_messages.py

```diff
@@ -11,10 +11,12 @@
 
 
 def error_no_hub_or_login_on_input(entity_type="IoT Hub"):
-    return (
-        "Please provide an {0} entity name (via the '{1}' or '-n' parameter)"
-        " or {0} connection string via --login..."
-    ).format(entity_type, PARAM_MAPPING[entity_type])
+    if entity_type:
+        return (
+            "Please provide an {0} entity name (via the '{1}' or '-n' parameter)"
+            " or {0} connection string via --login..."
+        ).format(entity_type, PARAM_MAPPING[entity_type])
+    return ("Please provide either a hub name or a DPS name.")
```

Comment:
Entity type is not supposed to be optional / accepting of `None`. If `None` is provided than a `ValueError` can be thrown.